core/src/main/scala/org/bitcoins/core/gcs/SipHashKey.scala

```diff
@@ -0,0 +1,17 @@
+package org.bitcoins.core.gcs
+import scodec.bits.ByteVector
+case class SipHashKey(bytes: ByteVector) {
+  require(
+    bytes.size == 16,
+    "Can only use a key length of 16 bytes, got: " + bytes.size)
+
+  def toArray: Array[Byte] = {
```

Comment:
nit: since this will be called from a `key: SipHashKey`, I'd recommend naming this `toByteArray` since it isn't obvious what `toArray` might mean for a `key`. (this can be done super easily by right clicking on this def in intellij and going to refactor > rename)

Reply:
You should be able to get rid of this method completely and just call `sipHashKey.bytes` if you extend `NetworkElement` on the case class.